test/tablet.py

```diff
@@ -613,14 +619,15 @@ def wait_for_mysqlctl_socket(self, timeout=60.0):
       timeout = utils.wait_step('waiting for socket files: %s' % str(wait_for),
                                 timeout, sleep_time=2.0)
 
-  def _add_dbconfigs(self, args, repl_extra_flags=None):
+  def _add_dbconfigs(self, cfg, args, repl_extra_flags=None):
     if repl_extra_flags is None:
       repl_extra_flags = {}
-    config = dict(self.default_db_config)
-    if self.keyspace:
-      config['app']['dbname'] = self.dbname
-      config['repl']['dbname'] = self.dbname
-    config['repl'].update(repl_extra_flags)
+    config = dict(cfg)
+    if 'repl' in config:
```

Comment:
Then this is a half-baked and hackish solution, because existence of config['app'] is not related to existence of config['repl'] in any way. It would be great to extract that into a separate "if".